src/encoded/helpers/upload_credentials.py

```diff
@@ -0,0 +1,181 @@
+"""
+Helper class to create user credentials to upload files to s3
+"""
+import copy
+import json
+
+import boto3
+import botocore
+
+
+EXTERNAL_BUCKET_STATEMENTS = [
+    {
+        'Action': 's3:GetObject',
+        'Resource': lambda s: 'arn:aws:s3:::%s/*' % s,
+        'Effect': 'Allow',
+    },
+    {
+        'Action': 's3:ListBucket',
+        'Resource': lambda s: 'arn:aws:s3:::%s' % s,
+        'Effect': 'Allow',
+    },
+    {
+        'Action': 's3:GetObjectAcl',
+        'Resource': lambda s: 'arn:aws:s3:::%s/*' % s,
+        'Effect': 'Allow',
+    }
+]
+
+
+def _compile_statements_from_list(buckets_list):
+    statements = []
+    if buckets_list:
+        for ext_policy in EXTERNAL_BUCKET_STATEMENTS:
+            new_policy = copy.copy(ext_policy)
+            new_policy['Resource'] = []
+            for line in buckets_list:
+                line = line.strip()
+                if line:
+                    line = line.strip()
+                    new_policy['Resource'].append(ext_policy['Resource'](line))
+            statements.append(new_policy)
+    return statements
+
+
+def _save_policy_json(policy_json, file_path):
+    with open(file_path + '.json', 'w') as file_handler:
+        json.dump(policy_json, file_handler)
+
+
+def _build_external_bucket_json(file_path):
+    try:
+        with open(file_path) as file_handler:
```

Comment:
is this file just a list of buckets?   It seems like this might be better handled by pyramid config rather that parsing a text file, even a simple one.

Reply:
I agree but I don't see how since we don't have private config. 

A previous project I worked on had base configs in the application repo and then pulled a config file tailored for the specific deployment from a private repo.  There is a need to have base configs in the application repo for adopters of the code base.  But also a need to have separate, specific config for us internally.  Like a list of which buckets we're sharing.

Reply:
oh right the names can't be in the repo